### ISSUE TITLE
Using my own images (Real data)

### ISSUE THREAD

**Issue by yuvalH9** [teddykoker/unsupervised-deep-homography#9]
Hi,
First of all thanks for the nice implementation using Pytorch - very helpful!

I want to test the model on real data, given two images of the same scene taken from two different angles.

What changes will be needed in the test file to work with such data in which the Homography is not generated synthetically?

What values should I use for the corners and how the Homography H will be estimated using the model output delta_hat in this case?

Thanks,
Yuval

**Comment by teddykoker** [comment]
Hi Yuval,

For the corners you should just be able to use:
```python
torch.tensor([[0, 0], [w, 0], [w, h], [0, h]])
```
Where `w` and `h` are the width and height of the image. Adding `delta_hat` (from the model) will give you the warped corners which can then be used with the original corners to obtain the homography matrix. Hope this helps!

I will warn you that I have not been able to get great results (its also been a while since I've worked on this), and it may require training some more on your dataset. If you are working on image alignment, another technique i have had work well is using RANSAC with matched features from a pretrained ResNet.

-Teddy

**Comment by yuvalH9** [comment]
Hi Teddy,

Thanks for your answer.

Just want to be sure, when you referring to the image width and height, you mean the patch coordinates?

Lets say the original images are 256 x 256 and the model (as given) can be feeded with a patch from the image with size 128 x 128.

So the value of `corners` should be the corners of the patch in a 256 x 256 coordinate system, or should I resize the original image to 128 x 128 and than use `corners = torch.tensor([[0, 0], [w, 0], [w, h], [0, h]])` where `w=h=128` ?

Thanks,
Yuval

**Comment by teddykoker** [comment]
I see, since the model was trained on images in a 128x128 coordinate system, you should just use `w=h=128`. The model doesn't actually use these corners in any way, it will just give you the movement of each corner, making the assumption that the input image is 128x128. In order to convert these corner perturbations back into a 256x256 system you will have to scale up the deltas and / or multiply the computed homography matrix by another matrix that does the scaling. Whatever you end up doing I would recommend testing your method with a pair of images with a known homography to ensure your pipeline is correct

**Comment by yuvalH9** [comment]
Well, I will describe exactly what I'm doing, let me know what should I do differently.

First of all I'm using a known dataset for Homography testing named [Image Matching: Graffiti](http://kahlan.eps.surrey.ac.uk/featurespace/web/data.htm), the dataset includes different scenes, for each scene we got 6 images taken from different angles, also we got the ground truth Homography between the images.

Those images are `800 x 640`, so first I do a center crop to create the maximal image (while keeping the same aspect ratio), i.e. getting a `640 x 640` images. Before putting the two image to the NN I resize the images into `128 x 128` to fit the input requirements (including normalization). As for the coroners values, I'm using the same values you stated before. In that case because the two images undergo the same resize I think that it will not be necessary to multiply the result Homography by another scaling matrix.
Unfortunately, the result for the matrix `H` are significantly different from the `H_GT`.

Do you think that I need to to something differently?

Regards,
Yuval
